temporalio/lib/temporalio/internal/worker/workflow_instance.rb

```diff
@@ -107,6 +107,10 @@ def initialize(details)
           end
           @query_handlers = HandlerHash.new(details.definition.queries, Workflow::Definition::Query)
           @update_handlers = HandlerHash.new(details.definition.updates, Workflow::Definition::Update)
+          @definition_options = Workflow::DefinitionOptions.new(
+            failure_exception_types: details.definition.failure_exception_types,
```

Comment:
This needs to replace the `@failure_exception_types` attribute and logic above I think. Specifically I think keeping `@worker_failure_exception_types` separate from workflow-level ones would be best and then just check both at failure check time